### PR TITLE
Bump psalm to v5 for intersection types support.

### PR DESCRIPTION
**Pull request by ADmad**

### CODE REVIEW

File: psalm-baseline.xml [hunk 1]
```diff
@@ -15,36 +12,72 @@
...
-      <code>$request</code>
-    </ArgumentTypeCoercion>
-  </file>
-  <file src="src/Routing/Middleware/RoutingMiddleware.php">
-    <ArgumentTypeCoercion occurrences="2">
-      <code>$request</code>
-      <code>$request</code>
-    </ArgumentTypeCoercion>
+  <file src="src/Core/Exception/CakeException.php">
+    <MethodSignatureMustProvideReturnType occurrences="62">
```
Comment (othercorey) [inline psalm-baseline.xml:17]: oof, talk about unmaintainable configs. can only generate this.

File: src/Collection/CollectionTrait.php [hunk 2]
```diff
@@ -1015,14 +1015,14 @@ public function countKeys(): int
      * Unwraps this iterator and returns the simplest
      * traversable that can be used for getting the data out
      *
-     * @return iterable
+     * @return \Iterator|array
```
Comment (othercorey) [inline src/Collection/CollectionTrait.php:1018]: Should we change this in 4.x?

File: src/Console/CommandCollection.php [hunk 3]
```diff
@@ -143,7 +145,7 @@ public function get(string $name): CommandInterface|string
      * Implementation of IteratorAggregate.
      *
      * @return \Traversable
-     * @psalm-return \Traversable<string, \Cake\Console\CommandInterface|class-string>
+     * @psalm-return \Traversable<string, \Cake\Console\CommandInterface|class-string<\Cake\Console\CommandInterface>>
```
Comment (othercorey) [inline src/Console/CommandCollection.php:148]: also in 4.x?

**Comment by othercorey** [comment]
Looks ok. Does it not work in 4.x though?

File: psalm-baseline.xml (hunk 1 above)
Comment (ADmad) [reply psalm-baseline.xml:17]: This looks like a bug in psalm to me. It isn't reading the narrowed down return type in `@method` annotation correctly.

**Comment by ADmad** [comment]
> Does it not work in 4.x though?

It should but personally I am not keen on devoting time on getting psalm v5 passing for 4.x.

**Pull request merged**